src/gui/application.cpp

```diff
@@ -141,18 +141,31 @@ bool Application::configVersionMigration()
 
     // Did the client version change?
     // (The client version is adjusted further down)
-    bool versionChanged = configFile.clientVersionString() != MIRALL_VERSION_STRING;
+    const auto versionChanged = configFile.clientVersionString() != MIRALL_VERSION_STRING;
 
     // We want to message the user either for destructive changes,
     // or if we're ignoring something and the client version changed.
-    bool warningMessage = !deleteKeys.isEmpty() || (!ignoreKeys.isEmpty() && versionChanged);
+    auto warningMessage = !deleteKeys.isEmpty() || (!ignoreKeys.isEmpty() && versionChanged);
 
-    if (!versionChanged && !warningMessage)
+    if (!versionChanged && !warningMessage) {
         return true;
+    }
 
-    const auto backupFile = configFile.backup();
+    // back up all old config file
+    QStringList backupFilesList;
+    QDir directory(configFile.configPath());
+    const auto anyConfigFileList = directory.entryInfoList({"*.cfg"}, QDir::Files);
+    for (const auto &file : anyConfigFileList) {
+        const auto fileName = file.fileName();
+        backupFilesList.append(configFile.backup(fileName));
+        if (file.baseName() != APPLICATION_CONFIG_NAME) {
```

Comment:
since you're renaming from `fileName` to `configFile.configFile()`, doesn't it make sense to change the condition
`if (file.baseName() != APPLICATION_CONFIG_NAME)` -> `if (fileName != configFile.configFile())` ?